MBBSEmu/IO/FileUtility.cs

```diff
@@ -34,55 +47,45 @@ public string FindFile(string modulePath, string fileName)
             {
-                var fileNameElements = fileName.Split(_directorySpecifier);
+                pathComponents.Enqueue(pathComponent);
+            }
 
-                //We only support 1 directory deep.. for now
-                if (fileNameElements.Length > 2 || fileNameElements.Length == 0)
-                    return fileName;
+            string fullPath = SearchPath(modulePath, pathComponents);
+            if (fullPath == null)
+            {
+                _logger?.Info($"Unable to find {fileName} under {modulePath}");
+                return null;
+            }
+            return Path.GetRelativePath(modulePath, fullPath);
+        }
 
-                fileNameElements[0] = fileNameElements[0].ToUpper();
-                fileNameElements[1] = fileNameElements[1].ToUpper();
-                if (File.Exists($"{modulePath}{fileNameElements[0]}{_directorySpecifier}{fileNameElements[1]}"))
-                    return string.Join(_directorySpecifier, fileNameElements);
+        private string SearchPath(string currentPath, Queue<string> pathComponents)
+        {
+            string component = pathComponents.Dequeue();
+            if (pathComponents.Count == 0)
+            {
+                return FindByEnumeration(currentPath, component, Directory.EnumerateFiles);
+            }
 
-                fileNameElements[0] = fileNameElements[0].ToLower();
-                fileNameElements[1] = fileNameElements[1].ToUpper();
-                if (File.Exists($"{modulePath}{fileNameElements[0]}{_directorySpecifier}{fileNameElements[1]}"))
-                    return string.Join(_directorySpecifier, fileNameElements);
+            // recurse into the next directory
+            string found = FindByEnumeration(currentPath, component, Directory.EnumerateDirectories);
+            return found == null ? null : SearchPath(found, pathComponents);
+        }
 
-                fileNameElements[0] = fileNameElements[0].ToUpper();
-                fileNameElements[1] = fileNameElements[1].ToLower();
-                if (File.Exists($"{modulePath}{fileNameElements[0]}{_directorySpecifier}{fileNameElements[1]}"))
-                    return string.Join(_directorySpecifier, fileNameElements);
+        private delegate IEnumerable<string> EnumerateFilesystemObjects(string path, string search, EnumerationOptions enumerationOptions);
 
-                fileNameElements[0] = fileNameElements[0].ToLower();
-                fileNameElements[1] = fileNameElements[1].ToLower();
-                if (File.Exists($"{modulePath}{fileNameElements[0]}{_directorySpecifier}{fileNameElements[1]}"))
-                    return string.Join(_directorySpecifier, fileNameElements);
+        private string FindByEnumeration(string root, string filename, EnumerateFilesystemObjects enumerateDelegate) {
+            foreach (string file in enumerateDelegate(root, filename, CASE_INSENSITIVE_ENUMERATION_OPTIONS))
```

Comment:
can be simplified to 

`private string FindByEnumeration(string root, string filename, EnumerateFilesystemObjects enumerateDelegate) => enumerateDelegate(root, filename, CASE_INSENSITIVE_ENUMERATION_OPTIONS).FirstOrDefault();`